core/thread.go

```diff
@@ -608,6 +628,95 @@ func (t *Thread) pushUpdate(index BlockInfo) {
 	})
 }
 
+// readable returns whether or not this thread is readable from the
+// perspective of the given address
+func (t *Thread) readable(addr string) bool {
+	if addr == t.initiator {
+		return true
+	}
+	switch t.ttype {
+	case repo.PrivateThread:
+		return false // should not happen
+	case repo.ReadOnlyThread:
+		return t.member(addr)
+	case repo.PublicThread:
+		return t.member(addr)
+	case repo.OpenThread:
+		return t.member(addr)
+	default:
+		return false
+	}
+}
+
+// annotatable returns whether or not this thread is annotatable from the
+// perspective of the given address
+func (t *Thread) annotatable(addr string) bool {
+	if addr == t.initiator {
+		return true
+	}
+	switch t.ttype {
+	case repo.PrivateThread:
+		return false // should not happen
+	case repo.ReadOnlyThread:
+		return false
+	case repo.PublicThread:
+		return t.member(addr)
+	case repo.OpenThread:
+		return t.member(addr)
+	default:
+		return false
+	}
+}
+
+// writable returns whether or not this thread can accept files from the
+// perspective of the given address
+func (t *Thread) writable(addr string) bool {
+	if addr == t.initiator {
+		return true
+	}
+	switch t.ttype {
+	case repo.PrivateThread:
+		return false // should not happen
+	case repo.ReadOnlyThread:
+		return false
+	case repo.PublicThread:
+		return false
+	case repo.OpenThread:
+		return t.member(addr)
+	default:
+		return false
+	}
+}
+
+// shareable returns whether or not this thread is shareable from one address to another
+func (t *Thread) shareable(from string, to string) bool {
+	switch t.sharing {
+	case repo.NotSharedThread:
+		return false
+	case repo.InviteOnlyThread:
+		return from == t.initiator && t.member(to)
```

Comment:
for this, can the initiator add members after the fact? i can't find that here? if no, then this seems quite limiting?

Reply:
see comment below